BaikalIO/file_utils.cpp

```diff
@@ -0,0 +1,150 @@
+#include <experimental/filesystem>
+#include <deque>
+#include <algorithm>
+
+namespace Baikal
+{
+    inline bool is_dot(std::experimental::filesystem::path const &path)
+    {
+        std::string const &filename = path.string();
+        return filename.size() == 1 && filename[0] == '.';
+    }
+
+    inline bool is_dotdot(std::experimental::filesystem::path const &path)
+    {
+        std::string const &filename = path.string();
+        return filename.size() == 2 && filename[0] == '.' && filename[1] == '.';
+    }
+
+    bool FindFilenameFromCaseInsensitive(std::string const &req_filename, std::string &actual_filename, bool resolve_symlinks = true)
+    {
+        const std::experimental::filesystem::path file_path(req_filename);
+
+        // First, check if file with required filename exists
+        if (std::experimental::filesystem::exists(file_path))
```

Comment:
This will not compile with latest compilers (MSVC 2017, GCC-8).

Reply:
Why? I've tested on both and all is fine.